packages/ast-spec/src/declaration/TSImportEqualsDeclaration/spec.ts

```diff
@@ -19,7 +19,45 @@ export interface TSImportEqualsDeclaration extends BaseNode {
    * import F3 = require('mod');
    * ```
    */
-  moduleReference: EntityName | TSExternalModuleReference;
-  // TODO(#1852) - breaking change remove this as it is invalid
+  moduleReference: Identifier | TSQualifiedName | TSExternalModuleReference;
```

Comment:
Slight tightening: `EntityName` is a moving target and doesn't really make sense here. I will continue to fix uses of it. It broke when `ThisExpression` became a valid `EntityName`.